{"version":3,"file":"ErrorBoundary.js","sourceRoot":"","sources":["../../core/util/ErrorBoundary.tsx"],"names":[],"mappings":";AAAA,OAAO,KAA4B,MAAM,OAAO,CAAC;AACjD,OAAO,EAAE,YAAY,EAAE,MAAM,cAAc,CAAC;AAW5C;IAAmC,iCAAoD;IAInF,uBAAY,KAAY;QAAxB,YACI,kBAAM,KAAK,CAAC,SAEf;QADG,KAAI,CAAC,KAAK,GAAG,EAAE,SAAS,EAAE,IAAI,EAAE,CAAC;;IACrC,CAAC;IAEQ,yCAAiB,GAA1B,UAA2B,KAAY,EAAE,SAA0B;QAC/D,IAAM,SAAS,GAAG,YAAY,CAAC,KAAK,EAAE,4BAA4B,EAAE;YAChE,eAAe,EAAE,SAAS,CAAC,cAAc;SAC5C,CAAC,CAAC;QACH,IAAI,CAAC,QAAQ,CAAC,EAAE,SAAS,WAAA,EAAE,CAAC,CAAC;IACjC,CAAC;IAEQ,8BAAM,GAAf;QACI,OAAO,IAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,MAAM,CAAC,IAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,QAAQ,IAAI,IAAI,CAAC;IACxG,CAAC;IAjBM,yBAAW,GAAG,eAAe,CAAC;IAC9B,0BAAY,GAA0B,EAAE,MAAM,EAAE,cAAM,OAAA,IAAI,EAAJ,CAAI,EAAE,CAAC;IAiBxE,oBAAC;CAAA,AAnBD,CAAmC,KAAK,CAAC,aAAa,GAmBrD;SAnBY,aAAa","sourcesContent":["import React, { PropsWithChildren } from \"react\";\r\nimport { captureError } from \"./error-util\";\r\nimport { Exception } from \"../Exception\";\r\n\r\ninterface Props {\r\n    render: (exception: Exception) => React.ReactElement | null;\r\n}\r\n\r\ninterface State {\r\n    exception: Exception | null;\r\n}\r\n\r\nexport class ErrorBoundary extends React.PureComponent<PropsWithChildren<Props>, State> {\r\n    static displayName = \"ErrorBoundary\";\r\n    static defaultProps: Pick<Props, \"render\"> = { render: () => null };\r\n\r\n    constructor(props: Props) {\r\n        super(props);\r\n        this.state = { exception: null };\r\n    }\r\n\r\n    override componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {\r\n        const exception = captureError(error, \"@@framework/error-boundary\", {\r\n            extraStacktrace: errorInfo.componentStack,\r\n        });\r\n        this.setState({ exception });\r\n    }\r\n\r\n    override render() {\r\n        return this.state.exception ? this.props.render(this.state.exception) : this.props.children || null;\r\n    }\r\n}\r\n"]}